hapi-loader/src/main/java/com/arkhn/hapi/loader/ResourceConsumer.java

```diff
@@ -38,16 +41,33 @@ public ResourceListener resourceListener() {
         @Autowired
         FhirContext myFhirContext;
 
+        @Autowired
+        private KafkaProducer producer;
+        private Logger LOGGER = LoggerFactory.getLogger(KafkaProducer.class);
+
         @KafkaHandler
         public void listen(String message) {
+            String fhirObject;
+            String batchId;
+            try {
+                // TODO I think we could find a better way to do that. With classes? With kafka?
+                JSONObject jsonObject = new JSONObject(message);
+                fhirObject = jsonObject.getString("fhir_object");
+                batchId = jsonObject.getString("batch_id");
```

Comment:
follow the guide:
https://medium.com/@iroshan.du/kafka-producer-consumer-with-batch-listener-with-customized-object-type-implementation-in-spring-29214a81aeb2